packages/shared/lib/core/wallet/utils/buildNftOutputData.ts

```diff
@@ -1,15 +1,26 @@
-import { convertDateToUnixTimestamp } from '@core/utils'
-import { FeatureTypes, UnlockConditionTypes } from '@iota/types'
+import { convertDateToUnixTimestamp, Converter } from '@core/utils'
+import type { FeatureTypes, UnlockConditionTypes } from '@iota/types'
 import type { BuildNftOutputData } from '@iota/wallet'
-import { ADDRESS_TYPE_ED25519, UNLOCK_CONDITION_ADDRESS, UNLOCK_CONDITION_EXPIRATION } from '../constants'
+import {
+    ADDRESS_TYPE_ED25519,
+    FEATURE_TYPE_METADATA,
+    FEATURE_TYPE_TAG,
+    UNLOCK_CONDITION_ADDRESS,
+    UNLOCK_CONDITION_EXPIRATION,
+    UNLOCK_CONDITION_STORAGE_DEPOSIT_RETURN,
+} from '../constants'
 import { convertBech32ToHexAddress } from './convertBech32ToHexAddress'
 
 export function buildNftOutputData(
     expirationDate: Date,
     nftId: string,
     immutableFeatures: FeatureTypes[],
+    metadata: string,
+    tag: string,
     recipientAddress: string,
-    accountAddress: string
+    accountAddress: string,
+    storageDeposit: string,
+    giftStorageDeposit: boolean
```

Comment:
I don't like this, the multitude of arguments should be converted into a couple of meaningful objects. Check Clean Code p.40 (https://github.com/dev-marko/clean-code-book/blob/master/Clean%20Code%20(%20PDFDrive.com%20).pdf)

![image](https://user-images.githubusercontent.com/45079109/199462730-a0a0a3f5-83af-4510-b037-933edfe9a590.png)

Reply:
i agree that a lot of arguments are hard to read, but i'm not sure how you would simplify this?

now in this case we don't need it anymore, because the function is now unused, but how would you minimize the args?